### PR TITLE
feat(builder): validate vue-app dependencies and suggest fix

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

**Background:** Nuxt requires all peer like dependencies as a dependency for easier usage. Sometimes this causes unwanted behaviors when users explicitly add a specific version in their `package.json` which is incompatible with nuxt.

This could also help to resolve popular `Vue packages version mismatch` error (#198, #669, #1084, #1414, #1851, #2079, #2406, #3454) and also regressions like #4668 that `vue-router@^3.0.2` is required by Nuxt and has fixes but users may encounter problems if anyhow they are still using `vue-router@3.0.0`.

Example error:
image

## Future improvements

- Automate dependency installs

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Review by clarkdo — APPROVED** [review]
LGTM

File: packages/vue-app/src/index.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-  'components/nuxt-loading.vue',
-  'components/nuxt-child.js',
-  'components/nuxt-link.server.js',
-  'components/nuxt-link.client.js',
-  'components/nuxt.js',
-  'components/no-ssr.js',
-  'views/app.template.html',
-  'views/error.html'
-]
+export const template = {
```
Comment (clarkdo) [inline packages/vue-app/src/index.js]: Should this export be default ?

Comment (pi0) [reply packages/vue-app/src/index.js]: Yes. But it breaks `yarn dev` script as we do `export * from '@nuxt/vue-app'`. Also reserving namespace for other top-level keys like `version` and `manifest`

Comment (clarkdo) [reply packages/vue-app/src/index.js]: gotcha

**Pull request merged**